cvat-core/src/frames.ts

```diff
@@ -707,80 +444,93 @@ export async function getFrame(
     isPlaying: boolean,
     step: number,
     dimension: DimensionType,
-) {
+    getChunk: (chunkNumber: number, quality: 'original' | 'compressed') => Promise<ArrayBuffer>,
+): Promise<FrameData> {
     if (!(jobID in frameDataCache)) {
-        const blockType = chunkType === 'video' ? cvatData.BlockType.MP4VIDEO : cvatData.BlockType.ARCHIVE;
+        const blockType = chunkType === 'video' ? BlockType.MP4VIDEO : BlockType.ARCHIVE;
         const meta = await serverProxy.frames.getMeta('job', jobID);
-        meta.deleted_frames = Object.fromEntries(meta.deleted_frames.map((_frame) => [_frame, true]));
-        const mean = meta.frames.reduce((a, b) => a + b.width * b.height, 0) / meta.frames.length;
+        const updatedMeta = {
+            ...meta,
+            deleted_frames: Object.fromEntries(meta.deleted_frames.map((_frame) => [_frame, true])),
+        };
+        const mean = updatedMeta.frames.reduce((a, b) => a + b.width * b.height, 0) / updatedMeta.frames.length;
         const stdDev = Math.sqrt(
-            meta.frames.map((x) => (x.width * x.height - mean) ** 2).reduce((a, b) => a + b) /
-                meta.frames.length,
+            updatedMeta.frames.map((x) => (x.width * x.height - mean) ** 2).reduce((a, b) => a + b) /
+            updatedMeta.frames.length,
         );
 
         // limit of decoded frames cache by 2GB
-        const decodedBlocksCacheSize = Math.floor(2147483648 / (mean + stdDev) / 4 / chunkSize) || 1;
-
+        const decodedBlocksCacheSize = Math.min(
+            Math.floor((2048 * 1024 * 1024) / ((mean + stdDev) * 4 * chunkSize)) || 1, 10,
+        );
```

Comment:
Can you please explain these calculations?

Reply:
Average number of pixels (mean) * number of channels per image (4) * chunks size.
This is maximum number of decoded chunks we can keep in memory 2048 Mb

stdDev is 0 for images with the same size and > 0 for images with different size, also statistics characteristic